### PR TITLE
Add KRM exec and Docker functions support

### PR DESCRIPTION
**Pull request by aabouzaid**
## Summary by CodeRabbit

* **New Features**
  * Added support for running as a Kubernetes Resource Model (KRM) function, enabling both exec and containerized modes for secret generation with Kustomize 4.0.0+.
  * Introduced Dockerfile and containerization support for streamlined deployment.
  * Provided comprehensive example directories and manifests for using environment variables and HashiCorp Vault as secret sources in both exec and Docker modes.

* **Bug Fixes**
  * Improved parsing of secret literals to correctly handle values containing "=" characters.

* **Documentation**
  * Expanded and reorganized README and added detailed example READMEs covering KRM function usage, legacy plugin migration, and troubleshooting.
  * Added documentation for new usage patterns, security considerations, and migration guidance from legacy plugins.

* **Tests**
  * Introduced integration and parsing tests for KRM function support, secret generation, and edge cases.

* **Chores**
  * Updated Go version and dependencies for improved compatibility and functionality.
  * Added `.dockerignore` to optimize Docker build context.
  * Updated GitHub Actions workflow for improved CI setup.

### CODE REVIEW

**Comment by aabouzaid** [comment]
@coderabbitai review

File: cmd/secretize/main.go [hunk 1]
```diff
@@ -2,20 +2,40 @@ package main
 
 import (
 	"fmt"
+	"io/ioutil"
```
[the remark this answers is not in the document]
Comment (AhmedElsayed101) [reply cmd/secretize/main.go]: It was updated

File: go.mod [hunk 2]
```diff
@@ -1,23 +1,88 @@
 module github.com/bbl/secretize
 
-go 1.13
+go 1.19
```
[the remark this answers is not in the document]
Comment (AhmedElsayed101) [reply go.mod]: It's updated

File: examples/exec/vault/secret-generator.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  - provider: hashicorp-vault
+    literals:
+      # Single secret values from docker-app namespace (with extra 'data' subfolder)
+      - DATABASE_URL=secret/data/data/docker-app/database-url:value
+      - API_KEY=secret/data/data/docker-app/api-key:value
+      - JWT_SECRET=secret/data/data/docker-app/jwt-secret:value
+    kv:
+      # KV secrets with all key-value pairs (with extra 'data' subfolder)
+      - secret/data/data/docker-app/app-config
+      - secret/data/data/docker-app/feature-flags
```
[the remark this answers is not in the document]
Comment (AhmedElsayed101) [reply examples/exec/vault/secret-generator.yaml:19]: No need, as Docker Compose is creating secrets with this path.

File: README.md [hunk 4]
```diff
@@ -30,6 +30,31 @@ It is possible to use multiple providers at once.
...
+
+#### Containerized KRM Function
+Use the Docker image (no installation required):
+```yaml
+# In your kustomization, reference the container image
+annotations:
+  config.kubernetes.io/function: |
+    container:
+      image: ghcr.io/bbl/secretize:latest
+```
```
[the remark this answers is not in the document]
Comment (AhmedElsayed101) [reply README.md]: Not used yet

File: examples/legacy/env/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    metadata:
+      labels:
+        app: example
+    spec:
+      containers:
+      - name: app
+        image: nginx:latest
+        envFrom:
+        - secretRef:
+            name: env-secrets
```
[the remark this answers is not in the document]
Comment (AhmedElsayed101) [reply examples/legacy/env/deployment.yaml]: Done.

File: examples/docker/vault/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    spec:
+      containers:
+      - name: app
+        image: nginx:latest
+        envFrom:
+        - secretRef:
+            name: docker-env-secrets
+        env:
+        - name: RUNTIME
+          value: "container"
```
[the remark this answers is not in the document]
Comment (AhmedElsayed101) [reply examples/docker/vault/deployment.yaml]: Done.

File: examples/legacy/vault/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    spec:
+      containers:
+      - name: app
+        image: nginx:latest
+        envFrom:
+        - secretRef:
+            name: docker-env-secrets
+        env:
+        - name: RUNTIME
+          value: "container"
```
[the remark this answers is not in the document]
Comment (AhmedElsayed101) [reply examples/legacy/vault/deployment.yaml]: Done.

**Comment by aabouzaid** [comment]
I'm closing this as the changes have been merged into the [upstream project](https://github.com/bbl/secretize).
